modules/smart-agent_logstash/conf/01-events_in_high.yaml

```diff
@@ -0,0 +1,23 @@
+module: logstash
+name: events in high
+
+transformation: ".mean(over='10m')"
+aggregation: true
+
+signals:
+  signal:
+    metric: node.stats.events.events.in
+    rollup: delta
+rules:
+  major:
+    description: is high
+    threshold: 25000
```

Comment:
as this detector thresholds highly depends on each environment, I would let threshold undefined to force the user to customize this according to his needs.

this can applied to all other detectors except may be the critical too low ones (equals to 0).

Reply:
I want to the module to be as useful out of the box as much as possible without too much configuration. With your suggestion the end user will get error when doing `terraform plan/apply` and that's very un-friendly.

Reply:
I agree about the goal to make all modules useful out the box as much as possible and this is why I suggest this in addition to my other suggestion to prefer relative based detectors (like percentage).

There are 2 possibilities:

- either the thresholds you choose have an universal or common logic so you can keep it but you just need to explain them (as Notes in readme, or as tip on detector, or mix them). e.g. on aws rds, the max connection is calculated depending on the instance type so we could specify a default value in this case which should match and work for any user who did not set a custom limit on their RDS.
- or these thresholds have been choose from your environment, with your requirements, your needs and your criteria so they have no sens or legitimacy for others and so they should not have been set as default in a generic module

the modules should be as plug and play as possible, meaning a user can deploy detectors without to customize anything and it works, users are used to this.

if you let default values for thresholds which must *always* be customized this is misleading for the user and a trap because they will not notice the problem until the moment they do not get alert when they want to.

if you do not set default values, so these variables will be automatically added to the module example usage in the readme to inform the user these variables are mandatory and yes if they forget them they will get an explicit and understandable error to force them to set, so no bad surprise in future because they deployed detectors which will never work in their environment (in my opinion this is more un-friendly)